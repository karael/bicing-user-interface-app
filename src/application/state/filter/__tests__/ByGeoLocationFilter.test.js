import ByGeoLocationFilter from 'application/state/filter/ByGeoLocationFilter';

describe('application/state/filter/ByGeoLocationFilter', () => {
  test('it can create a ByGeoLocationFilter from raw values', () => {
    const latitude = 41.3244;
    const longitude = 2.345;
    const limit = 5000;

    const filter = ByGeoLocationFilter.fromRawValues(latitude, longitude, limit);

    expect(filter.latitude).toEqual(latitude);
    expect(filter.longitude).toEqual(longitude);
    expect(filter.limit).toEqual(limit);
    expect(filter.zoom()).toEqual(12);
  });
  test('it can return a zoom value from a ByGeoLocationFilter if its limit is under 1000', () => {
    const filter = ByGeoLocationFilter.fromRawValues(40.321144, 2.2222, 200);

    expect(filter.zoom()).toEqual(15);
  });
  test('it can not create a filter if latitude is not a number between -90 and 90', () => {
    expect(() => ByGeoLocationFilter.fromRawValues(91, 2.2222, 200))
      .toThrowErrorMatchingSnapshot();
    expect(() => ByGeoLocationFilter.fromRawValues(-91, 2.2222, 200))
      .toThrowErrorMatchingSnapshot();
  });
  test('it can not create a filter if longitude is not a number between -180 and 180', () => {
    expect(() => ByGeoLocationFilter.fromRawValues(40.321144, 181, 200))
      .toThrowErrorMatchingSnapshot();
    expect(() => ByGeoLocationFilter.fromRawValues(40.321144, -181, 200))
      .toThrowErrorMatchingSnapshot();
  });
  test('it can not create a filter if longitude is not a number between 0 and 10000', () => {
    expect(() => ByGeoLocationFilter.fromRawValues(40.321144, 2.2222, 10001))
      .toThrowErrorMatchingSnapshot();
    expect(() => ByGeoLocationFilter.fromRawValues(40.321144, 2.2222, -1))
      .toThrowErrorMatchingSnapshot();
  });
});
