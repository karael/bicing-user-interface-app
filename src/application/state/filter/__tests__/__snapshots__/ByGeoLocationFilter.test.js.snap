// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application/state/filter/ByGeoLocationFilter it can not create a ByGeoLocationFilter if latitude is not a number between -90 and 90 1`] = `
"{
  \\"longitude\\": 2.2222,
  \\"limit\\": 200,
  \\"latitude\\" [31m[1][0m: 91
}
[31m
[1] \\"latitude\\" must be less than or equal to 90[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a ByGeoLocationFilter if latitude is not a number between -90 and 90 2`] = `
"{
  \\"longitude\\": 2.2222,
  \\"limit\\": 200,
  \\"latitude\\" [31m[1][0m: -91
}
[31m
[1] \\"latitude\\" must be larger than or equal to -90[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a ByGeoLocationFilter if longitude is not a number between -180 and 180 1`] = `
"{
  \\"latitude\\": 40.321144,
  \\"limit\\": 200,
  \\"longitude\\" [31m[1][0m: 181
}
[31m
[1] \\"longitude\\" must be less than or equal to 180[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a ByGeoLocationFilter if longitude is not a number between -180 and 180 2`] = `
"{
  \\"latitude\\": 40.321144,
  \\"limit\\": 200,
  \\"longitude\\" [31m[1][0m: -181
}
[31m
[1] \\"longitude\\" must be larger than or equal to -180[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a ByGeoLocationFilter if longitude is not a number between 0 and 10000 1`] = `
"{
  \\"latitude\\": 40.321144,
  \\"longitude\\": 2.2222,
  \\"limit\\" [31m[1][0m: 10001
}
[31m
[1] \\"limit\\" must be less than or equal to 10000[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a ByGeoLocationFilter if longitude is not a number between 0 and 10000 2`] = `
"{
  \\"latitude\\": 40.321144,
  \\"longitude\\": 2.2222,
  \\"limit\\" [31m[1][0m: -1
}
[31m
[1] \\"limit\\" must be larger than or equal to 0[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a filter if latitude is not a number between -90 and 90 1`] = `
"{
  \\"longitude\\": 2.2222,
  \\"limit\\": 200,
  \\"latitude\\" [31m[1][0m: 91
}
[31m
[1] \\"latitude\\" must be less than or equal to 90[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a filter if latitude is not a number between -90 and 90 2`] = `
"{
  \\"longitude\\": 2.2222,
  \\"limit\\": 200,
  \\"latitude\\" [31m[1][0m: -91
}
[31m
[1] \\"latitude\\" must be larger than or equal to -90[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a filter if longitude is not a number between -180 and 180 1`] = `
"{
  \\"latitude\\": 40.321144,
  \\"limit\\": 200,
  \\"longitude\\" [31m[1][0m: 181
}
[31m
[1] \\"longitude\\" must be less than or equal to 180[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a filter if longitude is not a number between -180 and 180 2`] = `
"{
  \\"latitude\\": 40.321144,
  \\"limit\\": 200,
  \\"longitude\\" [31m[1][0m: -181
}
[31m
[1] \\"longitude\\" must be larger than or equal to -180[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a filter if longitude is not a number between 0 and 10000 1`] = `
"{
  \\"latitude\\": 40.321144,
  \\"longitude\\": 2.2222,
  \\"limit\\" [31m[1][0m: 10001
}
[31m
[1] \\"limit\\" must be less than or equal to 10000[0m"
`;

exports[`application/state/filter/ByGeoLocationFilter it can not create a filter if longitude is not a number between 0 and 10000 2`] = `
"{
  \\"latitude\\": 40.321144,
  \\"longitude\\": 2.2222,
  \\"limit\\" [31m[1][0m: -1
}
[31m
[1] \\"limit\\" must be larger than or equal to 0[0m"
`;
